specification/trace/semantic_conventions/cloudevents.md

```diff
@@ -37,14 +39,13 @@ In case a CloudEvent is passed to the instrumented library with the Distributed
 
 ### Processing
 
-To trace the processing of one or more CloudEvents, instrumentation SHOULD create a new span.
-If a single event is processed, instrumentation SHOULD use the remote trace context from the Distributed Tracing Extension as a parent or MAY instead add it as a link on the processing span.
-
-If multiple events are processed together, for each event being processed, if the event has the Distributed Tracing Extension populated, instrumentation MUST add a link to the trace context from the extension on the processing span.
+When instrumented library supports processing of a single CloudEvent, instrumentation SHOULD create a new span to trace it. Instrumentation SHOULD set the remote trace context from the Distributed Tracing Extension as a link on the processing span.
```

Comment:
I liked the way the initial paragraph you wrote was structured. I think we should still layout recommendation for both cases - processing a single event and multiple, but use now Links as SHOULD. 

What do you think about this: 

> Instrumentation SHOULD create a new Span to trace the processing of one or more CloudEvents.

> When a single event is processed, instrumentation SHOULD set the remote trace context from the Distributed Tracing Extension as a link on the processing Span. 

> When multiple events are processed together (e.g. as a batch), for each event being processed, instrumentation SHOULD set the remote trace context from the Distributed Tracing Extension as a link on the processing span.

Reply:
with multiple events we have problems with attributes - we can't stamp event-specific attributes on multi-event-processing-span and I wanted to get away with a single event for now. We'd have to solve batching attributes for messaging and perhaps can reuse the same approach here?

Reply:
I guess on the related note: are you aware of any SDK that allows processing batch in a way that's auto-instrumentable?

Reply:
I don't know any from the top of my head. I think we can move forward without mentioning multiple-event processing and see what the others say.